CI Cucumber: Before: Request for simultaneously Before and not Before a date brings back that date

## Bug Report
A request was created for a soft contradiction where date should be generated for before and not before a date. The app generated the date as valid data, which contradicts both constraints applied.

> Scenario: 'before' run against a contradicting not 'before' should only only generate string, numeric and null
>      Given foo is before 2019-01-01T00:00:00.000
>        And foo is anything but before 2019-01-01T00:00:00.000
>        And the generator can generate at most 5 rows
>      Then the following data should be generated:
>        | foo                     |
>        | null                    |

### Steps to Reproduce:
 1. Run the above scenario in cucumber (or run the before.feature on branch 309-before-CI)
 2. See that the scenario **Scenario: 'before' run against a contradicting not 'before' should only only generate string, numeric and null** fails
 3. Review the error related to this failed test

### Expected Result:
Because of the soft contradiction, only null can be created.

### Actual Result:
The test fails and the error message shows that that the date 2019-01-01T00:00:00.000 is created:

> java.lang.AssertionError: 
> Expected: [null]
>      but: [2019-01-01T00:00:00.000], [null]
>  missing: 
>   counts: expected 1, but got 2
> 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
> 	at org.junit.Assert.assertThat(Assert.java:956)
> 	at org.junit.Assert.assertThat(Assert.java:923)
> 	at com.scottlogic.deg.generator.cucumber.steps.GeneralTestStep.theFollowingDataShouldBeGenerated(GeneralTestStep.java:160)
> 	at ✽.the following data should be generated:(C:/DataGenerator/data-engineering-generator/generator/src/test/java/com/scottlogic/deg/generator/cucumber/Before.feature:227)

### Environment:
Run in full generation mode via IntelliJ on branch 309-before-CI

### Additional Context:
![image](https://user-images.githubusercontent.com/43746958/52963431-9edef180-3397-11e9-89c7-bc4e14b84990.png)

--- a/generator/src/test/java/com/scottlogic/deg/generator/cucumber/Before.feature
+++ b/generator/src/test/java/com/scottlogic/deg/generator/cucumber/Before.feature
@@ -113,8 +113,7 @@ Scenario: not 'before' run against a non contradicting not 'before' should be su
        | 2019-01-01T00:00:00.002 |
        | 2019-01-01T00:00:00.003 |
        | 2019-01-01T00:00:00.004 |
-# Defect 613 "Request for simultaneously Before and not Before a date brings back that date" related to this scenario
-@ignore
+
 Scenario: 'before' run against a contradicting not 'before' should only only generate string, numeric and null
      Given foo is before 2019-01-01T00:00:00.000
        And foo is anything but before 2019-01-01T00:00:00.000

